pkg/auth/auth.go

```diff
@@ -13,9 +15,10 @@ import (
 )
 
 var (
-	validTTL   = time.Minute * 5
-	invalidTTL = time.Second * 30
-	cache      *AuthCache
+	validTTL     = time.Minute * 5
+	invalidTTL   = time.Second * 30
+	authEndpoint string
```

Comment:
to ensure this package works when flag.Parse() is never called, we need to initialize this to the default.
The, when the you define the flag variable you can use this var for the default value.

```
flag.StringVar(&authEndpoint, "auth-endpoint", authEndpoint, "Endpoint to authenticate users on")
```